src/main/resources/assets/carpet/scripts/draw_rc_1.sc

```diff
@@ -0,0 +1,196 @@
+import('math','_euclidean_sq','_vec_length');
+
+__config() -> {
+    'commands'->{
+        'sphere <center> <radius> <block>'->['draw_sphere',null, false],
+        'sphere <center> <radius> <block> replace <replacement>'->['draw_sphere',false],
+        'ball <center> <radius> <block>'->['draw_sphere', null, true],
+        'ball <center> <radius> <block> replace <replacement>'->['draw_sphere', true],
+        'diamond <center> <radius> <block>'->['draw_diamond',null],
+        'diamond <center> <radius> <block> replace <replacement>'->'draw_diamond',
+        'cone <center> <radius> <height> <pointing> <orientation> <block> <hollow>'->['draw_pyramid',null, false],
+        'cone <center> <radius> <height> <pointing> <orientation> <block> <hollow> replace <replacement>'->['draw_pyramid', false],
+        'pyramid <center> <radius> <height> <pointing> <orientation> <block> <hollow>'->['draw_pyramid',null, true],
+        'pyramid <center> <radius> <height> <pointing> <orientation> <block> <hollow> replace <replacement>'->['draw_pyramid', true],
+        'cylinder <center> <radius> <height> <orientation> <block> <hollow>'->['draw_prism', null, false],
+        'cylinder <center> <radius> <height> <orientation> <block> <hollow> replace <replacement>'->['draw_prism', false],
+        'cuboid <center> <radius> <height> <orientation> <block> <hollow>'->['draw_prism', null, true],
+        'cuboid <center> <radius> <height> <orientation> <block> <hollow> replace <replacement>'->['draw_prism', true]
+    },
+    'arguments'->{
+        'center'->{'type'->'pos', 'loaded'->'true'},
+        'radius'->{'type'->'int', 'suggest'->[], 'min'->0},//to avoid default suggestions
+        'replacement'->{'type'->'block'},
+        'height'->{'type'->'int', 'suggest'->[],'min'->0},
+        'orientation'->{'type'->'term', 'suggest'->['x','y','z']},
+        'pointing'->{'type'->'term','suggest'->['up','down']},
+        'hollow'->{'type'->'bool'}
+    },
+    'scope'->'global'
+};
+
+
+
+test(a,b,c)->print(center+ ' ' + radius + ' ' + block);
+
+//"Boilerplate" code
+
+set_block(x, y, z, block, replacement)-> (
+    success=0;
+    if(block != block(x, y, z),
+        set([x, y, z],block);
+        success+=1;
+
+        //todo add check for carpet setting CarpetSettings.fillUpdates
+        update(block(x, y, z))
```

Comment:
Scarpet's `set()` function already sends updates if `fillUpdates` is `true`, so removing the update() call is the same as checking fillUpdates (maybe even better, prevent double update).

Reply:
ah k nice, but Im afraid I just checked in the code and no bueno.

Reply:
@Ghoulboy78

Reply:
what?

Reply:
Sorry your comment didn't appear but the commit did, thought you didn't see it

Reply:
BTW `carpet_rules` is _way_ too expensive to be called for every single block being set (according to `profile_expr` in my computer, you can only run it 2100 times in a tick before taking >50ms, and that is without counting the rest of the code)